### DISCUSSION TITLE
BlueRetro Dreamcast Question on Remapping of D-Pad

### DISCUSSION THREAD

**Discussion by VirtualDavey** [darthcloud/BlueRetro#756] [Q&A] [answered]
I can only remap the Dreamcast D-Pad on a Switch Pro or PlayStation 5 controller only on itself. For example, I can program "D-Pad Up" to be instead "D-Pad Down" (on the Switch Pro's or PlayStation 5's D-Pad) with no issues. But I can't remap any of the D-Pad's directions to any other button. For example, I would like to program the "D-Pad Up" to the PlayStation 5's "R1 Trigger", but I cannot. When I do, 'D-Pad Up" no longer functions, either on the "R1 Trigger" or the D-Pad itself. I've attempted to remap "D-Pad Up" to various other buttons too with no success.

**Comment by darthcloud** [comment, marked as answer]
Sound like you are changing the source rather than the destination.

Maybe post a screenshot of what you did

**Reply by VirtualDavey** [reply]
I'm able to remap everything else with no issues, I'm playing Quake III Arena with dual sticks now and it's awesome. But I'm attempting to remap the "D-Pad" to the "R" trigger on the Switch Pro controller for Unreal Tournament (D-Pad up on the Dreamcast controller is actually "alternate fire" which is a horrible mapping for it), and Slave Zero (D-Pad Up is Jump, which is also horrible). Anyway, here's a picture (the highlighted rectangle is my attempted remap):
 https://i.postimg.cc/bw6p4K1H/DSC00123.jpg

**Comment by darthcloud** [comment]
you are using "right d-pad" which no BT controller have.
You want the one 4 line up.

"left d-pad up"

**Comment by darthcloud** [comment]
also you set your label backward,

source should be switch

destination is Dreamcast

**Reply by VirtualDavey** [reply]
Yes, I realize now after you mentioned it that the picture I had taken was showing the "Right D-Pad" being configured. That was probably an error while not paying attention while setting up for the pic. I'm fairly certain that I've used the "Left D-Pad" in my testings. But it is now working, so I believe having my "Source" and "Destination" backwards was the culprit. 
I just received this BlueRetro adapter yesterday and have been fiddling around with it. Am I correct that unless my Dreamcast has "Game ID" I can't have more than one remap setting saved? My Dreamcast Terraonion Mode doesn't have "Game ID" as far as I can tell. If there is a way to save multiple remap configurations, I'm unable to figure it out (and I've read the online user manual multiple times, but I'm not too tech savvy).

**Reply by darthcloud** [reply]
GameID exist only for N64, GC & PS1 so far. So you can't have multiple config on DC beside changing it between games.

**Reply by VirtualDavey** [reply]
Using the BlueRetro web config, is there an option to restore mapping to default? I see an option for "Factory Reset" but doesn't that also downgrade the firmware (I'm on v1.8.4 hw1 dreamcast at the moment).

**Reply by darthcloud** [reply]
In the preset page, load default_gamepad_kb preset
